lib/src/fields/form_builder_text_field.dart

```diff
@@ -410,7 +410,7 @@ class FormBuilderTextField extends FormBuilderField<String> {
               textDirection: textDirection,
               textCapitalization: textCapitalization,
               autofocus: autofocus,
-              readOnly: readOnly,
+              readOnly: state.readOnly,
```

Comment:
I presume that `state.readOnly` differs from `widget.readOnly` because the _Form_ could also be `readOnly`?

Reply:
Yes and moreover if the form is rebuilt with a new value for the `readonly` attribute, the widget will use the new value.

Reply:
But let's imagine that the form is readonly but a particular textfield should not be readonly with this change it's impossible! What do you think about it?

Reply:
The way `master` does this is that a _Field's State_ is `readOnly` when the _Field Widget_ is `readOnly` OR the _Form_ is `readOnly`.

This explains the subtle difference between `widget.readOnly` and `state.readOnly`.

In other words, **you cannot have a non-read-only Field on a read-only Form** because the Form's read-only attribute would make the Field read-only.

Reply:
So I change to `readOnly || state.readOnly`.

Reply:
I believe that is redundant because `State.readyOnly` is this:
```dart
  bool get readOnly => _formBuilderState?.readOnly == true || widget.readOnly;
```

`state.readOnly` is read-only if the _Form_ is read-only, OR the _Widget_ is read-only.

Your `readOnly` variable is synonymous with `widget.readOnly`.

The revised proposal would be this:
```dart
widget.readOnly || (_formBuilderState?.readOnly == true || widget.readOnly)
```
which is redundant.

I'm sorry for confusing you -- I think I am OK with your original change:
```dart
readOnly: state.readOnly,
```

Reply:
No worries 😂.

Ok perfect !